src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -94,6 +103,8 @@ trait Blockchain {
 
   def save(hash: ByteString, evmCode: ByteString): Unit
 
+  def save(node: MptNode): Unit
```

Comment:
I left this one outside blockchain intentionally. There are two main reasons to do so:
1. There are _2 implementations_ of node entities, the ones used while receiving them over the wire and the ones used by the MPT
2. Imo state isn't part of the blockchain. I like to think about state as a different part as it does not deal with blocks or how are those connected  :thinking: